suite-native/module-add-accounts/src/screens/AddCoinDiscoveryRunningScreen.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { networks, NetworkSymbol } from '@suite-common/wallet-config';
+import { VStack, Text, Spinner, SpinnerLoadingState } from '@suite-native/atoms';
+import { Screen } from '@suite-native/navigation';
+import { Translation } from '@suite-native/intl';
+import { Account } from '@suite-common/wallet-types';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    selectDeviceAccountsByNetworkSymbol,
+    selectDeviceDiscovery,
+} from '@suite-common/wallet-core';
+import { toggleEnabledDiscoveryNetworkSymbol } from '@suite-native/discovery';
+import { useCoinEnabling } from '@suite-native/coin-enabling';
+
+import { useAddCoinAccount } from '../hooks/useAddCoinAccount';
+
+export type AddCoinDiscoveryRunningScreenProps = {
+    networkSymbol: NetworkSymbol;
+};
+
+export const AddCoinDiscoveryRunningScreen = ({ route }) => {
+    const { networkSymbol, flowType } = route.params;
+    const dispatch = useDispatch();
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccountsByNetworkSymbol(state, networkSymbol),
+    );
+    const normalAccounts = accounts.filter(a => a.accountType === 'normal');
+    const nonEmptyAccounts = accounts.filter(a => !a.empty);
+
+    const discovery = useSelector(selectDeviceDiscovery);
+    const { navigateToAddCoinDiscoveryFinishedScreen, navigateToSuccessorScreen } =
+        useAddCoinAccount();
+    const { enabledNetworkSymbols, applyDiscoveryChanges } = useCoinEnabling();
+    const [loadingResult, setLoadingResult] = useState<SpinnerLoadingState>('idle');
+
+    const goToAccountDetail = ({ account }: { account: Account }) => {
+        navigateToSuccessorScreen({
+            flowType,
+            networkSymbol,
+            accountType: account.accountType,
+            accountIndex: account.index,
+        });
+    };
+
+    const handleFinish = () => {
+        if (!discovery) {
+            if (accounts.length > 0) {
+                setLoadingResult('success');
+            }
+
+            if (nonEmptyAccounts.length > 0 && normalAccounts.length > 0) {
+                navigateToAddCoinDiscoveryFinishedScreen(networkSymbol, flowType);
+            } else if (accounts.length > 0) {
+                goToAccountDetail({ account: normalAccounts[0] });
+            }
+        }
+    };
+
+    useEffect(() => {
+        if (
+            networkSymbol &&
+            !enabledNetworkSymbols.includes(networkSymbol) &&
+            accounts.length === 0 &&
+            !discovery
+        ) {
+            dispatch(toggleEnabledDiscoveryNetworkSymbol(networkSymbol));
+            applyDiscoveryChanges();
+        }
+
+        if (accounts.length > 0 && !discovery && loadingResult !== 'success') {
```

Comment:
```suggestion
        if (accounts.length > 0 && !discovery) {
```

IMO the last part of the check is not needed. WDYT?

Reply:
f471517693